File: frontend/web/src/models/Workflow.ts
import { Viewport, XYPosition } from 'reactflow';
import { PropertiesBase } from '@/nodes/PropertiesBase';

export interface NodeDto {
  id: string;
  type: string;
  position: XYPosition;
  positionAbsolute: XYPosition;
  data?: PropertiesBase;
}

export interface EdgeDto {
  id: string;
  source: string;
  target: string;
  sourceHandle: string;
  targetHandle: string;
}

export interface CreateWorkflowDto {
  title: string;
  description?: string;
  nodes: NodeDto[];
  edges: EdgeDto[];
  viewPort: Viewport;
  createdOn: string;
  modifiedOn: string;
}

export interface UpdateWorkflowDto {
  id: string;
  title: string;
  description: string;
  nodes: NodeDto[];
  edges: EdgeDto[];
  viewPort: Viewport;
  modifiedOn: string;
}

export interface Workflow {
  id: string;
  title: string;
  description: string;
  nodes: NodeDto[];
  edges: EdgeDto[];
  viewPort: Viewport;
  createdOn: string;
  modifiedOn: string;
}

File: services/WorkflowService/Entities/Workflow.cs
using MessagePack;
using MongoDB.Entities;

namespace WorkflowService;

public class Workflow : Entity
{
    public required string Title { get; set; }
    public required string Description { get; set; }
    public required List<Node> Nodes { get; set; }
    public required List<Edge> Edges { get; set; }
    public required ViewPort ViewPort { get; set; }
    public required string CreatedOn { get; set; }
    public required string ModifiedOn { get; set; }
}

[MessagePackObject]
public struct ViewPort
{
    public double X { get; set; }
    public double Y { get; set; }
    public double Zoom { get; set; }
}